db구조 수정, 컬렉션 추가

noah0316 opened the pull request:
- user.ts : 졸업생 취업 회사 관련 field 추가
- comment.ts : 댓글기능 구현을 위해 comment schema 추가
- homework.ts : 과제제출기능 구현을 위해 homework schema 추가하여 Pull Request 보냅니다😄

noah0316 committed:
Feat: Add Field to UserSchema
- user.ts : By adding a company field to UserSchema, you can see which company hired our alumni

noah0316 committed:
Feat: Add Homework, Comment Schema
- comment.ts : Add Comment Schema to add comments on homework
- homework.ts : Add Homework Schema to add homework submission feature

ndaemy reviewed (approved):
굿

ndaemy commented on src/models/comment.ts:
```diff
@@ -0,0 +1,23 @@
+  {
+    author: { type: String, required: true },
+    post: { type: Schema.Types.ObjectId, ref: 'Homework', required: true },
+  },
```
author에 왜 ref가 없죠??

noah0316 replied:
> author에 왜 ref가 없죠??

앗 ref에 대한 내용이 빠졌었네요!

ndaemy replied:
아니 FK면 type이 String이 아닐텐데?

noah0316 replied:
아 착각쓰... mongoose 익숙치 않아서 문서 좀 더 봐야겠다

ndaemy commented on src/models/homework.ts:
```diff
@@ -0,0 +1,27 @@
+  {
+    author: { type: String, required: true },
+    title: { type: String, required: true },
+    content: { type: String, required: true },
```
여기도 왜 author에 ref가 없죠?

ndaemy reviewed (changes requested):
흐음..

noah0316 committed:
Fix: Add collection name of document to reference

noah0316 committed:
Chore: Add husky hook, lint-staged

noah0316 committed:
Chore: Add Github Actions
- linter.yml: Add Github action for lint and typecheck when pushing to remote repository

ndaemy commented on .huskyrc:
```diff
@@ -0,0 +1,5 @@
+  "hooks": {
+    "pre-commit": "tsc && lint-staged"
+  }
+}
```
요거는 나중에 체크해보면 좋은데 파일 마지막에 빈 줄 하나가 필요해

ndaemy replied:
근데 .huskyrc가 필요해?

noah0316 replied:
@yuyaebean commit직전에 tsc실행 한 후에 lint-staged 실행시키려구요!!

ndaemy replied:
아 아니 그게 아니라 husky install 때리면 저 파일이 아닐거라서

아 그거 문제구나 husky 공식 README.md 가보면 v5를 받으라고 되어있는데 그건 yarn add -D husky@next 로 받아야돼

noah0316 replied:
아 맞네 공식문서 보니까 v4에서는 rc파일로 설정해줬었는데 v5에서는 .husky 폴더 하위에 pre-commit안에 정의해주는구나 내가 v4버전으로 했었구나..(라이브러리 사용할때 공식문서 보고 사용해야 하는 거 깨달음 블로그 3일 압수)

noah0316 committed:
Fix: Change Author Fieldtype

noah0316 committed:
Fix: Add empty line to last line of file

ndaemy reviewed (approved):
한번 찾아는 보소 뭔지

noah0316 committed:
Chore: Husky Migrate from v4 to v5

ndaemy commented on .husky/.gitignore:
```diff
@@ -0,0 +1 @@
+_
```
아 이거 나도 지금보고 알았네 _라는 폴더를 ignore해주는거

noah0316 replied:
저도 전에 왜 _라는 폴더 ignore하는지 궁금해서 뭐 있나 열어봤는데 쉘 script파일이 있더라구여

ndaemy reviewed (approved):
👏🏻
